ocaml/xapi/message_forwarding.ml

```diff
@@ -929,14 +929,18 @@ functor
         let self = Helpers.get_pool ~__context in
         let local_fn = Local.Pool.enable_tls_verification in
         let all_hosts = Xapi_pool_helpers.get_master_slaves_list ~__context in
-        Db.Pool.set_tls_verification_enabled ~__context ~self ~value:true ;
-        Cert_distrib.(
-          go ~__context ~existing_cert_strategy:Erase_old ~from_hosts:all_hosts
-            ~to_hosts:all_hosts) ;
-        all_hosts
-        |> List.iter (fun host ->
-               do_op_on ~local_fn ~__context ~host (fun session_id rpc ->
-                   Client.Pool.enable_tls_verification rpc session_id))
+
```

Comment:
My understanding is that users must have turned HA off before using `Pool.enable_tls_verification`. Is that correct?

If so, I would steal `assert_no_ha` and `assert_no_rpu` from xapi_psr.ml and use them here - I think `with_pool_operation` will only block if ha is _being_ enabled, not if it _is_ enabled (though it would be best to double-check). See https://github.com/xapi-project/xen-api/blob/a214de185c97178a6fa2c8a9d08a45624b3b6886/ocaml/xapi/xapi_psr.ml#L540